File: backend/internal/dtos/tts_request.go
package dtos

// TTSRequest represents a request to generate text-to-speech audio.
//
// Fields:
//   - Text (string): The text to be converted to speech.
//   - Rate (int): The speech rate.
//   - Voice (string): The voice to be used.
//   - NovelID (uint): The ID of the novel.
//   - ChapterNo (uint): The chapter number.
type TTSRequest struct {
	Text      string `json:"text"`
	Rate      int    `json:"rate"`
	Voice     string `json:"voice"`
	NovelID   uint   `json:"novelId"`
	ChapterNo uint   `json:"chapterNo"`
}

File: frontend/schemas/TTSRequest.ts
import { z } from 'zod';

const TTSRequestSchema = z.object({
  text: z.string(),
  voice: z.string(),
  rate: z.number(),
  novelId: z.number(),
  chapterNo: z.number(),
});

type TTSRequest = z.infer<typeof TTSRequestSchema>;
export { type TTSRequest, TTSRequestSchema };
